### ISSUE TITLE
Issue when showing multiple plots using WebGL

### ISSUE THREAD

**Issue by astrofrog** [bloomberg/bqplot#856]
I've run into an issue which manifests itself by the fact that when showing multiple figures that use WebGL (e.g. ScatterGL), one or more of the figures ends up empty. The following notebook reproduces this issue for me consistently in Firefox (but not Chrome):

https://gist.github.com/astrofrog/5434c4759bed440f98e2b84cbc6ab490

The issue is best demonstrated by selecting 'Restart Kernel and Run all cells'. If you do this repeatedly, every time (or almost) the first plot is empty:

![Screenshot from 2019-05-24 16-32-49](https://user-images.githubusercontent.com/314716/58339486-9c31e380-7e41-11e9-98d9-927f0c1a3c0d.png)

The console shows:

```
Error: WebGL warning: <SetDimensions>: Exceeded 16 live WebGL contexts for this principal, losing the least recently used one. three.module.js:21296
Error: WebGL warning: loseContext: Context is already lost. three.module.js:21296
Error: WebGL warning: <SetDimensions>: Exceeded 16 live WebGL contexts for this principal, losing the least recently used one. three.module.js:21296
Error: WebGL warning: loseContext: Context is already lost. three.module.js:21296
THREE.WebGLRenderer 91 three.module.js:21158
Error: WebGL warning: <SetDimensions>: Exceeded 16 live WebGL contexts for this principal, losing the least recently used one. three.module.js:21296
Error: WebGL warning: loseContext: Context is already lost.
```

**Comment by martinRenou** [comment]
I remember we discussed this with @maartenbreddels, we should consider not creating a WebGL context when it is not needed. That would delay this kind of issues.

Anyway, you would still have this issue if you create more than 16 ScatterGL plots, and I don't think we'll be able to go around it.

**Comment by maartenbreddels** [comment]
Yes, I was waiting for some PR's to be merged before working on this, but there are two issues here:
 * we shouldn't create the canvas when there are no gl marks
 * when we lose the context, we should make a screenshot of the canvas, and render that instead, similar to what pythreejs does (actually, it should be a library!)

**Comment by martinRenou** [comment]
> when we lose the context, we should make a screenshot of the canvas, and render that instead, similar to what pythreejs does

Is it really worth it? The plot won't respond to any interaction anymore

**Comment by maartenbreddels** [comment]
But it will go back to a WebGL canvas when that is needed, just turns static momentarily.
